nest/src/vault/NestBoringVaultModule.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import { Auth, Authority } from "@solmate/auth/Auth.sol";
+import { ERC20 } from "@solmate/tokens/ERC20.sol";
+import { FixedPointMathLib } from "@solmate/utils/FixedPointMathLib.sol";
+
+import { IAccountantWithRateProviders } from "../interfaces/IAccountantWithRateProviders.sol";
+import { IBoringVault } from "../interfaces/IBoringVault.sol";
+import { IComponentToken } from "../interfaces/IComponentToken.sol";
+import { console } from "forge-std/console.sol";
+
+/**
+ * @title NestBoringVaultModule
+ * @notice Base implementation for Nest vault modules
+ * @dev Implements common functionality for both Teller and AtomicQueue
+ */
+abstract contract NestBoringVaultModule is IComponentToken, Auth {
+
+    using SafeCast for uint256;
+    using FixedPointMathLib for uint256;
+
+    // Public State
+    IBoringVault public immutable vaultContract;
+    IAccountantWithRateProviders public immutable accountantContract;
+    uint256 public immutable decimals;
+    IERC20 public assetToken;
+
+    // Custom Errors
+    error InvalidReceiver();
+    error InvalidController();
+    error Unimplemented();
+    error InvalidAmount();
+    error ZeroOwner();
+    error InvalidOwner();
+    error ZeroVault();
+    error ZeroAccountant();
+    error ZeroEndpoint();
+    error ZeroAsset();
+    error InvalidMinimumMintPercentage();
+
+    // Constructor
+    constructor(address _owner, address _vault, address _accountant, IERC20 _asset) {
+        vaultContract = IBoringVault(_vault);
+        accountantContract = IAccountantWithRateProviders(_accountant);
+        decimals = vaultContract.decimals();
+        assetToken = _asset;
+    }
+
+    function setAsset(
+        IERC20 _asset
+    ) external requiresAuth {
+        assetToken = _asset;
+    }
+
+    function requestRedeem(
+        uint256 shares,
+        address controller,
+        address owner
+    ) public virtual override returns (uint256);
```

Comment:
Add `function deposit` and move `function requestRedeem` down with the rest of the functions?